### PR TITLE
HIVE-26397: Honour Iceberg sort orders when writing a table

### PR DESCRIPTION
**Pull request by lcspinter**
### What changes were proposed in this pull request?
Consider iceberg sort-order param when writing out data files. 

### How was this patch tested?
Unit test, manual test

### CODE REVIEW

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java [hunk 1]
```diff
@@ -682,7 +740,7 @@ public static Schema schema(Configuration config) {
   }
 
   /**
-   * Stores the serializable table data in the configuration.
+   * Stores the serializable table data in the configuration
```
Comment (pvary) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:743]: Why is this change?
BTW Javadoc expect `.` at the end of the first sentence... at least I remember somewhere fighting with adding extra dots at the end of the lines 😄

File: ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java [hunk 2]
```diff
@@ -346,6 +346,23 @@ default List<FieldSchema> acidSortColumns(org.apache.hadoop.hive.ql.metadata.Tab
...
+    return false;
+  }
+
+  /**
+   * Collect the columns that are used to sort the content of the data files
+   * @param table the table which is being sorted
+   * @return the list of columns that are used during data sorting
+   */
+  default List<FieldSchema> sortColumns(org.apache.hadoop.hive.ql.metadata.Table table) {
+    return Collections.emptyList();
```
Comment (pvary) [inline ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java:363]: Shall we throw an exception here, to find issues sooner?

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java [hunk 3]
```diff
@@ -334,24 +338,32 @@ public boolean supportsPartitionTransform() {
...
-      } else {
-        spec.setTransformType(PartitionTransformSpec.TransformType.valueOf(transformName));
-        spec.setTransformParam(Optional.empty());
-      }
+    return table.spec().fields().stream().map(f ->
+      getTransformSpec(table, f.transform().toString().toUpperCase(), f.sourceId())
+    ).collect(Collectors.toList());
+  }
+
+  private List<PartitionTransformSpec> getSortTransformSpec(Table table) {
```
[the remark this answers is not in the document]
Comment (lcspinter) [reply iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:346]: I will rename it to something more general, like `TransformSpec`.
Yes, we are able to handle every kind of sort-order definition.

**Review by szlta — APPROVED** [review]
Looks good to me. +1 pending tests

**Pull request merged**